test/library/idpdeployment.go

```diff
@@ -0,0 +1,209 @@
+package library
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apiserver/pkg/storage/names"
+	"k8s.io/client-go/kubernetes"
+
+	routev1 "github.com/openshift/api/route/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	routev1client "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+)
+
+func boolptr(b bool) *bool {
+	return &b
+}
+
+func deployPod(
+	t *testing.T,
+	clients *kubernetes.Clientset,
+	routeClient routev1client.RouteV1Interface,
+	name, image string,
+	env []corev1.EnvVar,
+	volumes []corev1.Volume,
+	volumeMounts []corev1.VolumeMount,
+	resources corev1.ResourceRequirements,
+) (namespace, host string, cleanup func()) {
+	testContext := context.TODO()
+
+	var err error
+	cleanup = func() {}
+
+	namespace = names.SimpleNameGenerator.GenerateName("e2e-test-authentication-operator-")
+	_, err = clients.CoreV1().Namespaces().Create(
+		testContext,
+		&corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   namespace,
+				Labels: CAOE2ETestLabels(), // add label to easily remove/get the NS by hand
+			},
+		},
+		metav1.CreateOptions{},
+	)
+	require.NoError(t, err)
+
+	cleanup = func() {
+		// remove the NS, it will take away all the resources create here along with it
```

Comment:
(No action required) Is this comment telling me anything I wouldn't know otherwise?

Reply:
it's a "don't panic and don't try to needlessly set up cleanups for the other resources" kind of comment, I like to remind myself of obvious stuff every now and then